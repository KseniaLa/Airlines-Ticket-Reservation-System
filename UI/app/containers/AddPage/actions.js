import {
  TICKETS_ADD_REQUESTED,
  TICKETS_ADD_SUCCEEDED,
  TICKETS_ADD_FAILED,
  CITIES_REQUESTED,
  CITIES_SUCCESS,
  CITIES_ERROR,
  COMPANIES_REQUESTED,
  COMPANIES_SUCCESS,
  COMPANIES_ERROR,
  DISCARD_STATE,
} from './constants';

export function addTicket(ticket) {
  return {
    type: TICKETS_ADD_REQUESTED,
    payload: ticket,
  };
}

export function addTicketSuccess() {
  return {
    type: TICKETS_ADD_SUCCEEDED,
  };
}

export function addTicketError() {
  return {
    type: TICKETS_ADD_FAILED,
  };
}

export function getAllCities(language) {
  return {
    type: CITIES_REQUESTED,
    payload: language,
  };
}

export function getAllCitiesSuccess(cities) {
  return {
    type: CITIES_SUCCESS,
    payload: cities,
  };
}

export function getAllCitiesError() {
  return {
    type: CITIES_ERROR,
  };
}

export function getAllCompanies(language) {
  return {
    type: COMPANIES_REQUESTED,
    payload: language,
  };
}

export function getAllCompaniesSuccess(companies) {
  return {
    type: COMPANIES_SUCCESS,
    payload: companies,
  };
}

export function getAllCompaniesError() {
  return {
    type: COMPANIES_ERROR,
  };
}

export function discardState() {
  return {
    type: DISCARD_STATE,
  };
}
