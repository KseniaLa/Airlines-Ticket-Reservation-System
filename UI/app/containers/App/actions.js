import { LOGOUT, SIGN_IN, SET_SEARCH, RESTORE, TRY_LOGIN } from './constants';

export function logout() {
  return {
    type: LOGOUT,
  };
}

export function tryLogin() {
  return {
    type: TRY_LOGIN,
  };
}

export function login(name, surname, isAdmin, token) {
  return {
    type: SIGN_IN,
    userName: name,
    userSurname: surname,
    isadmin: isAdmin,
    access_token: token,
  };
}

export function setSearch(from, to, date, flightClass) {
  return {
    type: SET_SEARCH,
    from,
    to,
    date,
    flightClass,
  };
}

export function restoreAuth() {
  return {
    type: RESTORE,
  };
}
