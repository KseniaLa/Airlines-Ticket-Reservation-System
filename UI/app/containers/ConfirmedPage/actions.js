import {
  CONFIRMATION_REQUESTED,
  CONFIRMATION_SUCCEEDED,
  CONFIRMATION_FAILED,
} from './constants';

export function confirmEmail(token) {
  return {
    type: CONFIRMATION_REQUESTED,
    payload: token,
  };
}

export function confirmEmailSuccess() {
  return {
    type: CONFIRMATION_SUCCEEDED,
  };
}

export function confirmEmailError() {
  return {
    type: CONFIRMATION_FAILED,
  };
}
