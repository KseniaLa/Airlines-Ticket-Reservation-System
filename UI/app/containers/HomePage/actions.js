import {
  RESULT_CITIES_REQUESTED,
  CITIES_FETCH_SUCCEEDED,
  CITIES_FETCH_FAILED,
} from './constants';

export function searchForCities() {
  return {
    type: RESULT_CITIES_REQUESTED,
  };
}

export function getCitiesSuccess(result) {
  return {
    type: CITIES_FETCH_SUCCEEDED,
    payload: result,
  };
}

export function getCitiesError() {
  return {
    type: CITIES_FETCH_FAILED,
  };
}
