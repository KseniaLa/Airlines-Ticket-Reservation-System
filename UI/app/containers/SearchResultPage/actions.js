import {
  RESULT_TICKETS_REQUESTED,
  TICKETS_FETCH_SUCCEEDED,
  TICKETS_FETCH_FAILED,
  DISCARD_DATA_READY,
  ADD_SUCCEEDED,
  ADD_FAILED,
  ADD_TICKET,
} from './constants';

export function searchForTickets(from, to, date, flightClass, lang) {
  return {
    type: RESULT_TICKETS_REQUESTED,
    payload: { from, to, date, flightClass },
    language: lang,
  };
}

export function getTicketsSuccess(result) {
  return {
    type: TICKETS_FETCH_SUCCEEDED,
    payload: result,
  };
}

export function getTicketsError() {
  return {
    type: TICKETS_FETCH_FAILED,
  };
}

export function discardDataReady() {
  return {
    type: DISCARD_DATA_READY,
  };
}

export function addTicket(ticket) {
  return {
    type: ADD_TICKET,
    payload: ticket,
  };
}

export function addTicketSuccess() {
  return {
    type: ADD_SUCCEEDED,
  };
}

export function addTicketError() {
  return {
    type: ADD_FAILED,
  };
}
