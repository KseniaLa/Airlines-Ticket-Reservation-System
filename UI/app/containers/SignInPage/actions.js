import {
  TRY_LOGIN,
  TRY_LOGIN_SUCCEEDED,
  TRY_LOGIN_FAILED,
  DISCARD_LOGIN_INFO,
  TRY_SIGNUP,
} from './constants';

export function tryLogin(uemail, upassword) {
  return {
    type: TRY_LOGIN,
    payload: { email: uemail, password: upassword },
  };
}

export function tryLoginSuccess(status) {
  return {
    type: TRY_LOGIN_SUCCEEDED,
    payload: status,
  };
}

export function tryLoginError() {
  return {
    type: TRY_LOGIN_FAILED,
  };
}

export function discardLogin() {
  return {
    type: DISCARD_LOGIN_INFO,
  };
}

export function trySignUp(uname, usurname, uemail, upassword) {
  return {
    type: TRY_SIGNUP,
    payload: {
      name: uname,
      surname: usurname,
      email: uemail,
      password: upassword,
    },
  };
}
