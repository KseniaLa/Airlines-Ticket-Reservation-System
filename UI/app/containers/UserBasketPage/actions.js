import {
  TICKETS_REQUESTED,
  TICKETS_FETCH_SUCCEEDED,
  TICKETS_FETCH_FAILED,
  DELETE_TICKET,
  ORDER_REQUESTED,
  ORDER_SUCCEEDED,
  ORDER_FAILED,
  DISCARD_ORDER_SUCCEEDED,
} from './constants';

export function getCartTickets(lang) {
  return {
    type: TICKETS_REQUESTED,
    payload: lang,
  };
}

export function getTicketsSuccess(result) {
  return {
    type: TICKETS_FETCH_SUCCEEDED,
    payload: result,
  };
}

export function getTicketsError() {
  return {
    type: TICKETS_FETCH_FAILED,
  };
}

export function deleteTicketFromCart(ticketId) {
  return {
    type: DELETE_TICKET,
    payload: ticketId,
  };
}

export function makeOrder() {
  return {
    type: ORDER_REQUESTED,
  };
}

export function makeOrderSuccess(result) {
  return {
    type: ORDER_SUCCEEDED,
    payload: result,
  };
}

export function makeOrderError() {
  return {
    type: ORDER_FAILED,
  };
}

export function discardOrderSucceeded() {
  return {
    type: DISCARD_ORDER_SUCCEEDED,
  };
}
