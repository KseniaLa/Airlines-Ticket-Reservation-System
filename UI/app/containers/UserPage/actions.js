import {
  HISTORY_REQUESTED,
  HISTORY_FETCH_SUCCEEDED,
  HISTORY_FETCH_FAILED,
} from './constants';

export function getUserIpHistory() {
  return {
    type: HISTORY_REQUESTED,
  };
}

export function getUserIpHistorySuccess(result) {
  return {
    type: HISTORY_FETCH_SUCCEEDED,
    payload: result,
  };
}

export function getUserIpHistoryError() {
  return {
    type: HISTORY_FETCH_FAILED,
  };
}
