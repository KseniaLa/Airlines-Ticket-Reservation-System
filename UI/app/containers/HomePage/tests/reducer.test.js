import { fromJS } from 'immutable';
import citiesReducer from '../reducer';
import { getCitiesSuccess, getCitiesError, resetCities } from '../actions';

describe('citiesReducer', () => {
  const initialState = fromJS({
    dataReceived: false,
    data: null,
  });
  it('returns the initial state', () => {
    expect(citiesReducer(undefined, {})).toMatchSnapshot();
  });

  it('sets', () => {
    expect(citiesReducer(initialState, getCitiesSuccess({}))).toMatchSnapshot();
  });

  it('handles', () => {
    expect(citiesReducer(initialState, getCitiesError())).toMatchSnapshot();
  });

  it('the', () => {
    expect(citiesReducer({}, resetCities())).toMatchSnapshot();
  });
});
