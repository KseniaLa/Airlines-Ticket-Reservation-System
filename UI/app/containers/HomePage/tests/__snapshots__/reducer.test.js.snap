// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`citiesReducer handles 1`] = `
Immutable.Map {
  "dataReceived": false,
  "data": null,
}
`;

exports[`citiesReducer returns the initial state 1`] = `
Immutable.Map {
  "dataReceived": false,
  "data": null,
}
`;

exports[`citiesReducer sets 1`] = `
Immutable.Map {
  "dataReceived": true,
  "data": Object {},
}
`;

exports[`citiesReducer the 1`] = `
Immutable.Map {
  "dataReceived": false,
  "data": null,
}
`;
